Promise/Promise+Extras.swift

```diff
@@ -24,25 +24,25 @@ extension Promise {
         })
     }
 
-    static func delay(delay: NSTimeInterval) -> Promise<()> {
+    static func delay(_ delay: TimeInterval) -> Promise<()> {
         return Promise<()>(work: { fulfill, reject in
             let nanoseconds = Int64(delay*Double(NSEC_PER_SEC))
-            let time = dispatch_time(DISPATCH_TIME_NOW, nanoseconds)
-            dispatch_after(time, dispatch_get_main_queue(), {
+            let time = DispatchTime.now() + Double(nanoseconds) / Double(NSEC_PER_SEC)
+            DispatchQueue.main.asyncAfter(deadline: time, execute: {
```

Comment:
Good call.  It's even simpler than that: `.asyncAfter(deadline: .now() + delay, …`.  I guess the migrator didn't catch that one.